File: compound interest.c
#include <stdio.h>
#include <math.h>

int main() {
    double principal, rate, time, amount, compound_interest;
    int n;

    // Accept user input
    printf("Compound Interest Calculator\n");

    printf("Enter the principal amount (P): ");
    scanf("%lf", &principal);

    printf("Enter the annual interest rate (in %%): ");
    scanf("%lf", &rate);

    printf("Enter the time (in years): ");
    scanf("%lf", &time);

    printf("Enter the number of times interest is compounded per year (n): ");
    scanf("%d", &n);

    // Calculate amount using the compound interest formula
    amount = principal * pow((1 + (rate / (100 * n))), (n * time));

    // Calculate compound interest
    compound_interest = amount - principal;

    // Display results
    printf("\nTotal amount after %.2lf years: $%.2lf\n", time, amount);
    printf("Compound Interest: $%.2lf\n", compound_interest);

    return 0;
}

File: compound interest.py
# Function to calculate compound interest
def calculate_compound_interest(principal, rate, time, n):
    # Convert rate from percentage to decimal
    rate_decimal = rate / 100
    # Calculate compound interest
    amount = principal * (1 + rate_decimal / n) ** (n * time)
    return amount

# Main program
def main():
    print("Compound Interest Calculator")
    
    # Accept user input
    principal = float(input("Enter the principal amount (P): "))
    rate = float(input("Enter the annual interest rate (in %): "))
    time = float(input("Enter the time (in years): "))
    n = int(input("Enter the number of times interest is compounded per year (n): "))
    
    # Calculate amount
    amount = calculate_compound_interest(principal, rate, time, n)
    
    # Calculate compound interest
    compound_interest = amount - principal
    
    # Display results
    print(f"\nTotal amount after {time} years: ${amount:.2f}")
    print(f"Compound Interest: ${compound_interest:.2f}")

# Run the main program
if __name__ == "__main__":
    main()
